www/profile/index.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+$servername = "localhost";
+$username = "worker";
+$password = "swiffcat";
+$db = "socialmedia";
+
+// Create connection
+$conn = new mysqli($servername, $username, $password, $db);
+
+// Check connection
+if ($conn->connect_error) {
+    header("HTTP/1.1 500 Internal Server Error");
+    die();
+}
+
+$headers = getallheaders();
+if (!array_key_exists("token", $headers)) {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+$token = getallheaders()["token"];
+
+// If invalid header token, die
+$stmt = $conn->prepare("SELECT user_id FROM auth WHERE token = ?");
+$stmt->bind_param("s", $a1);
+$a1 = $token;
+$stmt->execute();
+$stmt->store_result();
+$accessor_id = null;
+$stmt->bind_result($accessor_id);
+if ($stmt->fetch()) {
+    // user_id is bound
+} else {
+    header("HTTP/1.1 401 Unauthorized");
+    die();
+}
+
+if ($_SERVER['REQUEST_METHOD'] === 'GET') {
+    if (!isset($_GET["id"])) {
+        echo $_GET;
+        header("HTTP/1.1 403 Forbidden");
+        die();
+    }
+
+    $user_id = $_GET["id"];
+
+    // make sure the user exists and grab their data
+    $stmt = $conn->prepare("SELECT * FROM users WHERE user_id = ?");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $count = $result->num_rows;
+    if ($count != 1) {
+        header("HTTP/1.1 403 Forbidden");
+        die();
+    }
+    $row = $result->fetch_assoc();
+
+    // Retrieve data from the posts table
+    $stmt = $conn->prepare("SELECT * FROM posts WHERE user_id = ? ORDER BY created_at DESC");
+    $stmt->bind_param("s", $a1);
+    $a1 = $user_id;
+    $stmt->execute();
+    $result = $stmt->get_result();
+    $posts = array();
+
+    if ($result->num_rows > 0) {
+        while ($post_row = $result->fetch_assoc()) {
+            $posts[] = $post_row;
+        }
+    }
+
+    // add posts to the values obtained earlier
+    $row["posts"] = $posts;
+    $json = json_encode($row);
+    echo $json;
+
+    // Close the database connection
+    mysqli_close($conn);
+} else if ($_SERVER['REQUEST_METHOD'] === 'PUT') {
+
+    $inputData = file_get_contents('php://input');
+
+    // Split the input into parts
+    $boundary = substr($inputData, 0, strpos($inputData, "\r\n"));
+
+    // Split content by boundary and get rid of last -- element
+    $parts = array_slice(explode($boundary, $inputData), 1);
+    $_PUT = array();
+
+    foreach ($parts as $part) {
```

Comment:
This entire block is incredibly rigid and error prone. Incredibly! Looking at it you should see how this string parsing is easily broken and will only work in specific happy paths

Chatgpt gave me this while pipes the content into a file
```
if ($_SERVER['REQUEST_METHOD'] == 'PUT') {
    // Get the input stream
    $input = fopen('php://input', 'rb');

    // Create a temporary file to store the uploaded file
    $temp = tmpfile();

    // Read the data a chunk at a time and write to the temporary file
    while ($chunk = fread($input, 1024)) {
        fwrite($temp, $chunk);
    }

    // Done with the input file so we can close it
    fclose($input);

    // Now we move the pointer back to the start of the temp file
    rewind($temp);

    // Here you can do whatever you want with the temp file. For example, move it to a new location:
    $targetPath = '/path/to/your/directory/uploaded_file_name';
    if (rename(stream_get_meta_data($temp)['uri'], $targetPath)) {
        echo 'File was moved successfully';
    } else {
        echo 'Failed to move file';
    }

    // Close temp file
    fclose($temp);
} 
```

Reply:
I think you should do a POST unless you are going to invest a ton of time into understanding how to make this resilient

Reply:
I agree that I should probably just do a post. It would be nice to just use a while to read a single file thats given but we also are taking in a usersId to make sure that the correct user is updating their profile. So I think rather than going through the headache of trying to parse it I should just use the post.

The question then becomes, what should I do about the aboutMe section then? That is still currently a Put, I have to read both the id and aboutme so I have to parse it in some way, even if it was just the aboutMe I'd still have to parse it in some way.